### PR TITLE
Migrate image to fluentd.image and sumologic.setup.job.image

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

* migrate `image` to `sumologic.setup.job.image` and to `fluentd.image` (for now use the same image but prepare a separate image for setup job in a different PR)
* add this to v2 migration script and add its tests to CI
* change tests
* adjust migration script (v1) tests as they used `yq` arguments which changed in 3.4.0 and now need to be provided differently - we might actually end up remove this script as it's only relevant to 1.x branches
* bump `yq` in vagrant and on CI to 3.4.1

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Remove `image` from the `values.yaml`

File: tests/upgrade_script/static/prometheus_service_monitors.input.yaml [hunk 1]
```diff
@@ -1,59 +1,59 @@
 prometheus-operator:
```
Comment (sumo-drosiek) [inline tests/upgrade_script/static/prometheus_service_monitors.input.yaml:1]: I don't see a point of changing this file

Comment (pmalek-sumo) [reply tests/upgrade_script/static/prometheus_service_monitors.input.yaml]: Done, reverted

File: .travis.yml [hunk 2]
```diff
@@ -7,10 +7,10 @@ before_install:
...
   - gem install bundler
   # yq
-  - curl --retry 10 --retry-max-time 120 --retry-delay 5 -L --remote-name https://github.com/mikefarah/yq/releases/download/3.2.1/yq_linux_amd64
-  - sudo mv yq_linux_amd64 /usr/local/bin/yq-3.2.1
-  - sudo chmod +x /usr/local/bin/yq-3.2.1
-  - sudo ln -s /usr/local/bin/yq-3.2.1 /usr/local/bin/yq
+  - curl --retry 10 --retry-max-time 120 --retry-delay 5 -L --remote-name https://github.com/mikefarah/yq/releases/download/3.4.1/yq_linux_amd64
+  - sudo mv yq_linux_amd64 /usr/local/bin/yq-3.4.1
+  - sudo chmod +x /usr/local/bin/yq-3.4.1
+  - sudo ln -s /usr/local/bin/yq-3.4.1 /usr/local/bin/yq
```
Comment (sumo-drosiek) [inline .travis.yml:13]: please update vagrantfile as well

Comment (pmalek-sumo) [reply .travis.yml:13]: Done

**Review by sumo-drosiek — APPROVED** [review]
👌

**Pull request merged**